manifests/config.pp

```diff
@@ -65,9 +65,10 @@
   }
 
   git_config { $title:
-    key   => $_key,
-    value => $value,
-    user  => $user,
-    scope => $scope,
+    key     => $_key,
+    value   => $value,
+    user    => $user,
+    scope   => $scope,
+    require => Package['git'],
```

Comment:
Replace the line with `require => Package[$::git::package_name],`. That being said, we don't want to require that if this isn't being used to manage the git package, so maybe instead change the section to:
```puppet
$package = $::git::package_manage ? {
  true    => Package[$::git::package_name],
  default => undef,
}

git_config { $title:
  key     => $_key,
  value   => $value,
  user    => $user,
  scope   => $scope,
  require => $package,
}
```